### ISSUE TITLE
[RPi 3B/3B+] -o both param causes massive slowdown

### ISSUE THREAD

**Issue by nikkoman** [popcornmix/omxplayer#625]
-o both playback causes severe system slowdown on Stretch when RCA is used for A/V.

I've been through a lot of trial and error and mostly "blamed" ES but it wasn't the case.

It doesn't happen when HDMI is used and the temporary solution is to set -o to ALSA:HW:0,0.

**Comment by popcornmix** [comment]
Where do you want the audio to output from? The analogue/headphone socket?
Does ``-o local`` work?

**Comment by nikkoman** [comment]
I wanted to automatically have audio from both HDMI (when I use my HDTV) and analogue RCA (when I use a CRT for 240p gaming).

When it's set to both (via the EmulationStation menu where it's the default value for omxplayer audio) the whole system starts going through a severe slowdown on Stretch builds.

All this has been verified via shell where I executed videos alone and located the problem being the `- both` param.

This has been further discussed: https://retropie.org.uk/forum/topic/17107/bug-solved-omxplayer-on-stretch-based-builds-stutters-on-composite/

Local worked only for analogue audio but setting it to ALSA:HW;0,0 via ES did the trick. It behaves as if it was set to both on pre-stretch versions.

**Comment by popcornmix** [comment]
> -o both playback causes severe system slowdown on Stretch when RCA is used for A/V.

What do you mean by this? Is this a problem with the video/audio played, or high cpu observed while the video is playing?

**Comment by pjft** [comment]
From the "top" screenshots he shared, it doesn't seem to create any different CPU load, so it's more about the former.

**Comment by githubguy007** [comment]
Also seeing this issue on Raspbian Stretch (4.14.30+ #1102) running on  Pi Zero, Pi Zero W, and Pi 3 B+ hardware.  I am using composite video, and I am NOT running ES.

When "-o both" is used with omxplayer the video runs in slow motion (fractional play speed).  CPU utilization is no where near overburdened.  Changing to "-o hdmi" or "-o local"  normalizes video replay speed. 

/boot/config.txt 
sdtv_mode=0
sdtv_aspect=3
dtparam=audio=on

pi@Channel_One:~$ omxplayer -v
omxplayer - Commandline multimedia player for the Raspberry Pi
        Build date: Wed, 05 Jul 2017 16:40:39 +0100
        Version   : 5a25a57 [debian]
        Repository: git@github.com:XECDesign/omxplayer.git

**Comment by popcornmix** [comment]
Does adding ``audio_pwm_mode=0`` to config.txt and rebooting have any effect?

**Comment by githubguy007** [comment]
Yes, adding `audio_pwm_mode=0` to /boot/config.txt allows `omxplayer -o both` to playback video at full speed.

**Comment by nikkoman** [comment]
Very interesting. `audio_pwm_mode=0` does the job done! Not only it eliminates the issue with `-o both` but also stops some menu delay on ES (major slowdown aside). Why is thar required now? Are there any drawbacks from using that mode?

**Comment by popcornmix** [comment]
Ping @P33M

**Comment by whmountains** [comment]
I have a similar issue, which may be related.  `-o both` causes the audio to stutter terribly if the pi was started without a display connected.

`audio_pwm_mode=0` greatly improves the situation but doesn't fix it completely.

**Comment by GunZi200** [comment]
I would like to add that this issue also affects Raspberry Pi Zero. Had this issue almost a year ago and `audio_pwm_mode=0` did work but I found changing `omxplayer -o both -b --layer 10000 "$line"` to `omxplayer -b --layer 10000 "$line"` was much smoother. I tried a lot of things before I got this fixed, curious can take a look here: https://sudomod.com/forum/viewtopic.php?f=44&t=5953&start=10
